pkg/cluster/acsk/action/action_create_nodepool.go

```diff
@@ -0,0 +1,183 @@
+	log       logrus.FieldLogger
+	nodePools []*model.ACSKNodePoolModel
+	context   *ACKContext
+	region    string
+}
+
+// NewCreateACSKNodePoolAction creates a new CreateACSKNodePoolAction
+func NewCreateACSKNodePoolAction(log logrus.FieldLogger, nodepools []*model.ACSKNodePoolModel, clusterContext *ACKContext, region string) *CreateACSKNodePoolAction {
+	return &CreateACSKNodePoolAction{
+		log:       log,
+		nodePools: nodepools,
+		context:   clusterContext,
+		region:    region,
+	}
+}
+
+// GetName returns the name of this CreateACSKNodePoolAction
+func (a *CreateACSKNodePoolAction) GetName() string {
+	return "CreateACSKNodePoolAction"
+}
+
+// ExecuteAction executes this CreateACSKNodePoolAction
+func (a *CreateACSKNodePoolAction) ExecuteAction(input interface{}) (interface{}, error) {
+	cluster, ok := input.(*acsk.AlibabaDescribeClusterResponse)
+	if !ok {
+		return nil, errors.New("invalid input")
+	}
+
+	if len(a.nodePools) == 0 {
+		r, err := getClusterDetails(a.context.ClusterID, a.context.CSClient)
+		if err != nil {
+			return nil, emperror.With(err, "clusterName", cluster.Name)
+		}
+
+		return r, nil
+	}
+	a.log.Infoln("EXECUTE CreateACSKNodePoolAction, cluster name", cluster.Name)
+
+	errChan := make(chan error, len(a.nodePools))
+	instanceIdsChan := make(chan []string, len(a.nodePools))
+	defer close(errChan)
+	defer close(instanceIdsChan)
+
+	for _, nodePool := range a.nodePools {
+		go func(nodePool *model.ACSKNodePoolModel) {
+			scalingGroupRequest := ess.CreateCreateScalingGroupRequest()
+			scalingGroupRequest.SetScheme(requests.HTTPS)
+			scalingGroupRequest.SetDomain("ess." + cluster.RegionID + ".aliyuncs.com")
+			scalingGroupRequest.SetContentType(requests.Json)
+
+			a.log.WithFields(logrus.Fields{
+				"region":        cluster.RegionID,
+				"zone":          cluster.ZoneID,
+				"instance_type": nodePool.InstanceType,
+			}).Info("creating scaling group")
+
+			scalingGroupRequest.MinSize = requests.NewInteger(nodePool.MinCount)
+			scalingGroupRequest.MaxSize = requests.NewInteger(nodePool.MaxCount)
+			scalingGroupRequest.VSwitchId = cluster.VSwitchID
+			scalingGroupRequest.ScalingGroupName = fmt.Sprintf("asg-%s-%s", nodePool.Name, cluster.ClusterID)
+
+			createScalingGroupResponse, err := a.context.ESSClient.CreateScalingGroup(scalingGroupRequest)
+			if err != nil {
+				errChan <- emperror.WrapWith(err, "could not create Scaling Group", "nodePoolName", nodePool.Name, "clusterName", cluster.Name)
+				instanceIdsChan <- nil
+				return
+			}
+
+			nodePool.AsgId = createScalingGroupResponse.ScalingGroupId
+			a.log.Infof("Scaling Group with id %s successfully created", nodePool.AsgId)
+			a.log.Infof("Creating scaling configuration for group %s", nodePool.AsgId)
+
+			scalingConfigurationRequest := ess.CreateCreateScalingConfigurationRequest()
+			scalingConfigurationRequest.SetScheme(requests.HTTPS)
+			scalingConfigurationRequest.SetDomain("ess." + cluster.RegionID + ".aliyuncs.com")
```

Comment:
The api endpoint could be extracted into a const similar to `ossEndpointFmt = "https://oss-%s.aliyuncs.com"`